core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```
@@ -352,7 +356,7 @@ class AuthorizerIntegrationTest extends KafkaServerTestHarness {
   }
 
   @Test
-  def testConsumeWithNoTopicAccess() {
+  def testConsumeWithoutTopicDescribeAccess() {
```

The expectation of this test case is not too clear. It looks like the exception that we raise is actually being thrown from the offset fetch request, and it seems like it may be a separate bug that we don't handle the UNKNOWN_TOPIC_OR_PARTITION error code in the offset fetch response handler. Maybe we should fix that? Ultimately, until we have timeouts for some of the consumer's blocking calls, we may not have a great way to test this case.

---

Yes, but we can make the current behavior explicit, right? This test only passes because we do not handle the UNKNOWN_TOPIC_OR_PARTITION error in the offset fetch response handler. So:
1. In `ConsumerCoordinator.OffsetFetchResponseHnadler`, let's add an explicit case for the `UNKNOWN_TOPIC_OR_PARTITION`. I suggest we let the error be raised as an instance of `UnknownTopicOrPartitionException` instead of a generic `KafkaException`. We should also give the user a message which lets them know that they may not have Describe access on the topic.
2. Let's do the same for `ConsumerCoordinator.OffsetCommitResponseHandler`.
3. Let's change the name of this test case so that it's clear what exactly is expected to fail.

---

Hi 
sorry but could you confirm ... :
If `ConsumerCoordinator.OffsetFetchResponseHandler` raises an UTOP 
that is a retriable exception and `ConsumerCoordinator.fetchCommittedOffsets` remains stuck in the `while` loop.
The client would not see the exception. The unit test would have to be coded for a timeout. 
Is this the behavior you'd like to see ? With a log entry while stuck in the loop ?

---

@edoardocomar Hmm... Yeah, that's a fair point. How about we catch the error explicitly and throw as `KafkaException` with a message indicating the topic/partition is unknown which mentions that the the user may not have describe access.